### PR TITLE
DEVDOCS-6025 [update]: Get Prices (batch), add saved & reference_request fields

### PR DESCRIPTION
**Pull request by bc-andreadao**
# [DEVDOCS-6025]

## What changed?

* Add `saved` field & `reference_request` field. See ticket comments for slack conversation.

Screenshot proof for `reference_request`:
proof

## Release notes draft

- Corrected the response body of the [Get Prices (Batch)](https://developer.bigcommerce.com/docs/rest-management/pricing/products#get-prices-batch) endpoint to include the `saved` and `reference_request` fields. 

## Anything else?
@bigcommerce/team-pricing 

[DEVDOCS-6025]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-6025?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: reference/pricing.sf.yml [hunk 1]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                              tax_exclusive:
+                                type: number
+                                description: The estimated tax exclusive price for this product based on the provided customer group.
+                              tax_inclusive:
+                                type: number
+                                description: The estimated tax inclusive price for this product based on the provided customer group.
+                            description: The minimum advertised price (MAP) allowed to be shown on a storefront. A value supplied by the merchant and used for display purposes.
+                          saved:
+                            type: object
+                            description: The price that the Stencil context caches.
```
Comment (bc-andreadao) [inline reference/pricing.sf.yml:163]: Not sure if this description is correct

**Review by bc-vinbase — APPROVED** [review]
LGTM

**Review by megdesko — APPROVED** [review]
👍  looks good, thanks for the updates!

File: reference/pricing.sf.yml [hunk 2]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                            properties:
+                              as_entered:
+                                type: number
+                                description: The price provided by the merchant as entered in their catalog/price list; may include or exclude tax.
+                              entered_inclusive:
+                                type: boolean
+                                description: Determines whether the `as_entered` price is inclusive or exclusive of tax, based on the stores tax jurisdiction.
+                              tax_exclusive:
+                                type: number
+                                description: The estimated tax exclusive price for this product based on the provided customer group.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:156]:
```suggestion
                                description: The estimated tax-exclusive price for this product depends on the provided customer group.

```

File: reference/pricing.sf.yml [hunk 3]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                              tax_inclusive:
+                                type: number
+                                description: The estimated tax inclusive price for this product based on the provided customer group.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:159]:
```suggestion
                                description: The estimated tax-inclusive price for this product depends on the provided customer group.

```

File: reference/pricing.sf.yml [hunk 4]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                            description: The minimum advertised price (MAP) allowed to be shown on a storefront. A value supplied by the merchant and used for display purposes.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:160]:
```suggestion
                            description: The minimum advertised price (MAP) you can display on a storefront. A value supplied by the merchant and used for display purposes.

```

File: reference/pricing.sf.yml [hunk 5]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                          saved:
+                            type: object
+                            description: The amount that merchants save, determined by the difference between `retail_price` and `calculated_price`.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:163]:
```suggestion
                            description: The amount merchants can save is determined by the difference between `retail_price` and `calculated_price`.

```

File: reference/pricing.sf.yml [hunk 6]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                            properties:
+                              as_entered:
+                                type: number
+                                description: The price provided by the merchant as entered in their catalog/price list; may include or exclude tax.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:167]:
```suggestion
                                description: The price provided by the merchant, as entered in their catalog/price list, may include or exclude tax.

```

File: reference/pricing.sf.yml [hunk 7]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                              entered_inclusive:
+                                type: boolean
+                                description: Determines whether the `as_entered` price is inclusive or exclusive of tax, based on the stores tax jurisdiction.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:170]:
```suggestion
                                description: Determines whether the `as_entered` price is inclusive or exclusive of tax, based on the store's tax jurisdiction.

```

File: reference/pricing.sf.yml [hunk 8]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                              tax_exclusive:
+                                type: number
+                                description: The estimated tax exclusive price for this product based on the provided customer group.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:173]:
```suggestion
                                description: The estimated tax-exclusive price for this product depends on the provided customer group.

```

File: reference/pricing.sf.yml [hunk 9]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                              tax_inclusive:
+                                type: number
+                                description: The estimated tax inclusive price for this product based on the provided customer group.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:176]:
```suggestion
                                description: The estimated tax-inclusive price for this product depends on the provided customer group.

```

File: reference/pricing.sf.yml [hunk 10]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                          price:
+                            type: object
+                            properties:
+                              as_entered:
+                                type: number
+                                description: The price provided by the merchant as entered in their catalog/price list; may include or exclude tax.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:182]:
```suggestion
                                description: The price provided by the merchant, as entered in their catalog/price list, may include or exclude tax.

```

File: reference/pricing.sf.yml [hunk 11]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                              entered_inclusive:
+                                type: boolean
+                                description: Determines whether the `as_entered` price is inclusive or exclusive of tax, based on the stores tax jurisdiction.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:185]:
```suggestion
                                description: Determines whether the `as_entered` price is inclusive or exclusive of tax, based on the store's tax jurisdiction.

```

File: reference/pricing.sf.yml [hunk 12]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                              tax_exclusive:
+                                type: number
+                                description: The estimated tax exclusive price for this product based on the provided customer group.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:188]:
```suggestion
                                description: The estimated tax-exclusive price for this product depends on the provided customer group.

```

File: reference/pricing.sf.yml [hunk 13]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                              tax_inclusive:
+                                type: number
+                                description: The estimated tax inclusive price for this product based on the provided customer group.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:191]:
```suggestion
                                description: The estimated tax-inclusive price for this product depends on the provided customer group.

```

File: reference/pricing.sf.yml [hunk 14]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                            description: The merchant-entered price for a product could be including or excluding tax. Price must be defined when creating a product and serves as the default price.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:192]:
```suggestion
                            description: The merchant-entered price for a product could include or exclude tax. When creating a product, you must define the price, which serves as the default price.

```

File: reference/pricing.sf.yml [hunk 15]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                          calculated_price:
+                            type: object
+                            properties:
+                              as_entered:
+                                type: number
+                                description: The price provided by the merchant as entered in their catalog/price list; may include or exclude tax.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:198]:
```suggestion
                                description: The price provided by the merchant, as entered in their catalog/price list, may include or exclude tax.
```

File: reference/pricing.sf.yml [hunk 16]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                              entered_inclusive:
+                                type: boolean
+                                description: Determines whether the `as_entered` price is inclusive or exclusive of tax, based on the stores tax jurisdiction.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:201]:
```suggestion
                                description: Determines whether the `as_entered` price is inclusive or exclusive of tax, based on the store's tax jurisdiction.

```

File: reference/pricing.sf.yml [hunk 17]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                              tax_exclusive:
+                                type: number
+                                description: The estimated tax exclusive price for this product based on the provided customer group.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:204]:
```suggestion
                                description: The estimated tax-exclusive price for this product depends on the provided customer group.

```

File: reference/pricing.sf.yml [hunk 18]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                              tax_inclusive:
+                                type: number
+                                description: The estimated tax inclusive price for this product based on the provided customer group.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:207]:
```suggestion
                                description: The estimated tax-inclusive price for this product depends on the provided customer group.

```

File: reference/pricing.sf.yml [hunk 19]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                            description: The shopper price for a product including modifier, option, and option set rules. The `calculated_price` may include or exclude estimates for tax.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:208]:
```suggestion
                            description: The shopper price for a product includes modifier, option, and option set rules. The `calculated_price` may include or exclude estimates for tax.
```

File: reference/pricing.sf.yml [hunk 20]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                          price_range:
+                            type: object
+                            properties:
+                              minimum:
+                                type: object
+                                properties:
+                                  as_entered:
+                                    type: number
+                                    description: The price provided by the merchant as entered in their catalog/price list; may include or exclude tax.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:217]:
```suggestion
                                    description: The price provided by the merchant, as entered in their catalog/price list, may include or exclude tax.

```

File: reference/pricing.sf.yml [hunk 21]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                                  entered_inclusive:
+                                    type: boolean
+                                    description: Determines whether the `as_entered` price is inclusive or exclusive of tax, based on the stores tax jurisdiction.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:220]:
```suggestion
                                    description: Determines whether the `as_entered` price is inclusive or exclusive of tax, based on the store's tax jurisdiction.

```

File: reference/pricing.sf.yml [hunk 22]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                                  tax_exclusive:
+                                    type: number
+                                    description: The estimated tax exclusive price for this product based on the provided customer group.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:223]:
```suggestion
                                    description: The estimated tax-exclusive price for this product depends on the provided customer group.

```

File: reference/pricing.sf.yml [hunk 23]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                                  tax_inclusive:
+                                    type: number
+                                    description: The estimated tax inclusive price for this product based on the provided customer group.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:226]:
```suggestion
                                    description: The estimated tax-inclusive price for this product depends on the provided customer group.

```

File: reference/pricing.sf.yml [hunk 24]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                                description: The price for a product including estimates for tax.
+                              maximum:
+                                type: object
+                                properties:
+                                  as_entered:
+                                    type: number
+                                    description: The price provided by the merchant as entered in their catalog/price list; may include or exclude tax.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:233]:
```suggestion
                                    description: The price provided by the merchant, as entered in their catalog/price list, may include or exclude tax.

```

File: reference/pricing.sf.yml [hunk 25]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                                  entered_inclusive:
+                                    type: boolean
+                                    description: Determines whether the `as_entered` price is inclusive or exclusive of tax, based on the stores tax jurisdiction.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:236]:
```suggestion
                                    description: Determines whether the `as_entered` price is inclusive or exclusive of tax, based on the store's tax jurisdiction.

```

File: reference/pricing.sf.yml [hunk 26]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                                  tax_exclusive:
+                                    type: number
+                                    description: The estimated tax exclusive price for this product based on the provided customer group.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:239]:
```suggestion
                                    description: The estimated tax-exclusive price for this product depends on the provided customer group.

```

File: reference/pricing.sf.yml [hunk 27]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                                  tax_inclusive:
+                                    type: number
+                                    description: The estimated tax inclusive price for this product based on the provided customer group.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:242]:
```suggestion
                                    description: The estimated tax-inclusive price for this product depends on the provided customer group.

```

File: reference/pricing.sf.yml [hunk 28]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                                description: The price for a product including estimates for tax.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:243]:
```suggestion
                                description: The price for a product includes estimates for tax.

```

File: reference/pricing.sf.yml [hunk 29]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                            description: The minimum and maximum price that will typically apply to this product. Only used for complex products (products with variants).
+                          retail_price_range:
+                            type: object
+                            properties:
+                              minimum:
+                                type: object
+                                properties:
+                                  as_entered:
+                                    type: number
+                                    description: The price provided by the merchant as entered in their catalog/price list; may include or exclude tax.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:253]:
```suggestion
                                    description: The price provided by the merchant, as entered in their catalog/price list, may include or exclude tax.

```

File: reference/pricing.sf.yml [hunk 30]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                                  entered_inclusive:
+                                    type: boolean
+                                    description: Determines whether the `as_entered` price is inclusive or exclusive of tax, based on the stores tax jurisdiction.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:256]:
```suggestion
                                    description: Determines whether the `as_entered` price is inclusive or exclusive of tax, based on the store's tax jurisdiction.

```

File: reference/pricing.sf.yml [hunk 31]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                                  tax_exclusive:
+                                    type: number
+                                    description: The estimated tax exclusive price for this product based on the provided customer group.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:259]:
```suggestion
                                    description: The estimated tax-exclusive price for this product depends on the provided customer group.

```

File: reference/pricing.sf.yml [hunk 32]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                                  tax_inclusive:
+                                    type: number
+                                    description: The estimated tax inclusive price for this product based on the provided customer group.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:262]:
```suggestion
                                    description: The estimated tax-inclusive price for this product depends on the provided customer group.

```

File: reference/pricing.sf.yml [hunk 33]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                                description: The price for a product including estimates for tax.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:263]:
```suggestion
                                description: The price for a product includes estimates for tax.

```

File: reference/pricing.sf.yml [hunk 34]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                              maximum:
+                                type: object
+                                properties:
+                                  as_entered:
+                                    type: number
+                                    description: The price provided by the merchant as entered in their catalog/price list; may include or exclude tax.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:269]:
```suggestion
                                    description: The price provided by the merchant, as entered in their catalog/price list, may include or exclude tax.

```

File: reference/pricing.sf.yml [hunk 35]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                                  entered_inclusive:
+                                    type: boolean
+                                    description: Determines whether the `as_entered` price is inclusive or exclusive of tax, based on the stores tax jurisdiction.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:272]:
```suggestion
                                    description: Determines whether the `as_entered` price is inclusive or exclusive of tax, based on the store's tax jurisdiction.

```

File: reference/pricing.sf.yml [hunk 36]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                                  tax_exclusive:
+                                    type: number
+                                    description: The estimated tax exclusive price for this product based on the provided customer group.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:275]:
```suggestion
                                    description: The estimated tax-exclusive price for this product depends on the provided customer group.

```

File: reference/pricing.sf.yml [hunk 37]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                                  tax_inclusive:
+                                    type: number
+                                    description: The estimated tax inclusive price for this product based on the provided customer group.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:278]:
```suggestion
                                    description: The estimated tax-inclusive price for this product depends on the provided customer group.

```

File: reference/pricing.sf.yml [hunk 38]
```diff
@@ -273,12 +141,147 @@ paths:
...
+                                description: The price for a product including estimates for tax.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:279]:
```suggestion
                                description: The price for a product includes estimates for tax.

```

File: reference/pricing.sf.yml [hunk 39]
```diff
@@ -351,6 +365,28 @@ paths:
                 meta: {}
 components:
   schemas:
+    ReferenceRequest:
+      type: object
+      description: The original details of the request used to identify the exact variant of a product to fetch prices.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:370]:
```suggestion
      description: You can use the original details of the request to identify the exact product variant and fetch prices.

```

File: reference/pricing.sf.yml [hunk 40]
```diff
@@ -351,6 +365,28 @@ paths:
...
+      properties:
+        product_id:
+          type: integer
+          description: The (required) product ID of the item.
+        variant_id:
+          type: integer
+          description: The (optional) variant ID of the item.
+        options:
+          type: array
+          description: The optional product option configuration for which this price was generated.
```
Comment (bc-traciporter) [inline reference/pricing.sf.yml:380]:
```suggestion
          description: The optional product option configuration for this generated price.
```

**Review by bc-traciporter — COMMENTED** [review]
@bc-andreadao Sorry for all the comments. There is a lot of repetition. I wanted to change the tense to active voice but I hope I did not change the meaning. Feel free to accept the changes that feels right to you.

File: reference/pricing.sf.yml [hunk 41]
```diff
@@ -257,8 +125,8 @@ paths:
                                 tax_inclusive:
                                   type: number
                                   description: The estimated tax inclusive price for this product based on the provided customer group.
-                              description: The price for a product including estimates for tax.
-                            maximum:
+                              description: The (optional) RRP/retail price configured for this product. Used for price comparison and storefront display purposes.
```
[the remark this answers is not in the document]
Comment (bc-andreadao) [reply reference/pricing.sf.yml:128]: Thanks, I'll update the sentence to `The (optional) RRP/retail price configured for this product and used for price comparison and storefront display purposes` to preserve the meaning

**Pull request merged**